content/en/docs/migrating/_index.md

```diff
@@ -829,6 +899,355 @@ to wait for a {{% alias service=S3 %}} Bucket to exist, you must construct a `Bu
 [s3.BucketExistsWaiter]({{< apiref "service/s3#BucketExistsWaiter" >}}). The `s3.BucketExistsWaiter` provides a
 `Wait` method which can be used to wait for a bucket to become available.
 
+## Request customization
+
+The monolithic [request.Request]({{< apiref v1="aws/request#Request" >}}) API
+has been re-compartmentalized.
+
+### Operation input/output
+
+The opaque `Request` fields `Params` and `Data`, which hold the operation input
+and output structures respectively, are now accessible within specific
+middleware phases as input/output:
+
+Request handlers which reference `Request.Params` and `Request.Data` must be migrated to middleware.
+
+#### migrating `Params`
+
+```go
```

Comment:
has all the following new code examples been actually executed to some extent? i think it would be good to actually see this code successfully execute

Reply:
I dropped them into my IDE and found some minor issues which I've corrected. I didn't run them fully, though, they're not meant to be drop-in snippets, more examples of what v1 would look like vs v2.